File: iosApp/iosApp/ui/settlement/PreviewableSettlementMethodTruthSource.swift
//
//  PreviewableSettlementMethodTruthSource.swift
//  iosApp
//
//  Created by jimmyt on 10/3/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import SwiftUI

/**
 A `UISettlementMethodTruthSource` implementation used for previewing user interfaces
 */
class PreviewableSettlementMethodTruthSource: UISettlementMethodTruthSource {
    
    /**
     A list of containing the user's `SettlementMethods`.
     */
    var settlementMethods: [SettlementMethod] = []
    
    /**
     Not used since this class is for previewing user interfaces, but required for adoption of `UISettlementMethodTruthSource`.
     */
    func addSettlementMethod(
        settlementMethod: SettlementMethod,
        newPrivateData: PrivateData,
        stateOfAdding: Binding<AddingSettlementMethodState>,
        addSettlementMethodError: Binding<Error?>
    ) {}
    
    /**
     Not used since this class is for previewing user interfaces, but required for adoption of `UISettlementMethodTruthSource`.
     */
    func editSettlementMethod(
        settlementMethod: SettlementMethod,
        newPrivateData: PrivateData,
        stateOfEditing: Binding<EditingSettlementMethodState>,
        editSettlementMethodError: Binding<Error?>,
        privateDataBinding: Binding<PrivateData?>
    ) {}
    
    /**
     Not used since this class is for previewing user interfaces, but required for adoption of `UISettlementMethodTruthSource`.
     */
    func deleteSettlementMethod(
        settlementMethod: SettlementMethod,
        stateOfDeleting: Binding<DeletingSettlementMethodState>,
        deleteSettlementMethodError: Binding<Error?>
    ) {}
    
}

File: androidApp/src/main/java/com/commuto/interfacemobile/android/ui/settlement/PreviewableSettlementMethodTruthSource.kt
package com.commuto.interfacemobile.android.ui.settlement

import androidx.compose.runtime.MutableState
import androidx.compose.runtime.snapshots.SnapshotStateList
import com.commuto.interfacemobile.android.settlement.SettlementMethod
import com.commuto.interfacemobile.android.settlement.privatedata.PrivateData

/**
 * A [UISettlementMethodTruthSource] implementation used for previewing user interfaces.
 * @property settlementMethods A [SnapshotStateList] of containing the user's [SettlementMethod]s.
 */
class PreviewableSettlementMethodTruthSource: UISettlementMethodTruthSource {
    override var settlementMethods = SnapshotStateList<SettlementMethod>().also { snapshotStateList ->
        SettlementMethod.sampleSettlementMethodsEmptyPrices.map {
            snapshotStateList.add(it)
        }
    }

    /**
     * Not used since this class is for previewing user interfaces, but required for implementation of
     * [UISettlementMethodTruthSource].
     */
    override fun addSettlementMethod(
        settlementMethod: SettlementMethod,
        newPrivateData: PrivateData,
        stateOfAdding: MutableState<AddingSettlementMethodState>,
        addSettlementMethodException: MutableState<Exception?>
    ) {}

    /**
     * Not used since this class is for previewing user interfaces, but required for adoption of
     * [UISettlementMethodTruthSource].
     */
    override fun editSettlementMethod(
        settlementMethod: SettlementMethod,
        newPrivateData: PrivateData,
        stateOfEditing: MutableState<EditingSettlementMethodState>,
        editSettlementMethodException: MutableState<Exception?>,
        privateDataMutableState: MutableState<PrivateData?>
    ) {}
}